validity/src/proofs.rs

```diff
@@ -181,23 +186,50 @@ where
                 .serialize(&proof)
                 .unwrap();
 
-            reply = AggProofResponse {
-                success: true,
-                error: "".into(),
-                last_proven_block: req.last_proven_block,
-                end_block: end_block as u64,
-                proof_request_id: proof_bytes.into(),
+            let proved_op_request = OPSuccinctRequest {
+                proof: proof_bytes.into(),
+                ..op_request.clone()
             };
+
+            // Create an aggregation proof request to cover the range with the checkpointed L1 block hash.
+            let result = self
+                .proof_requester
+                .db_client
+                .insert_request(&proved_op_request)
+                .await
+                .map_err(|e| Status::internal(format!("Failed to save request to DB: {}", e)))?;
+
+            if result.rows_affected() > 0 {
+                // Fetch the last inserted ID
+                let last_id: i64 = sqlx::query_scalar!(
+                    r#"
+                SELECT currval(pg_get_serial_sequence('requests', 'id'))
+                "#
+                )
+                .fetch_one(&self.proof_requester.db_client.pool)
+                .await
+                .map_err(|e| Status::internal(format!("Failed to fetch agg proof ID: {}", e)))?
+                .unwrap();
```

Comment:
What does this `.unwrap()` do? Should it be passed to the caller instead?

Reply:
> `fetch_one` is returning `Result<Option<i64>, Error>` so need to unwrap here. Let me know if there's more idiomatic way 🙏

I was thinking of propagating it as an error (not sure if I get the method names right, just a sketch:
```suggestion
                .ok_or(Status::internal("last_proof_id not found"))?;
```

I understand that since you just inserted it into the database it should be there but my instinct would be to still prefer a non-panicking way of handling it.